server/src/domain/metadata/metadata.service.ts

```diff
@@ -566,18 +567,33 @@ export class MetadataService {
       return false;
     }
 
-    const sidecarPath = `${asset.originalPath}.xmp`;
-    const exists = await this.storageRepository.checkFileExists(sidecarPath, constants.R_OK);
-    if (exists) {
-      await this.assetRepository.save({ id: asset.id, sidecarPath });
+    // XMP sidecars can come in two filename formats. For a photo named photo.ext, the filenames are photo.ext.xmp and photo.xmp
+    const sidecarPathWithExt = `${asset.originalPath}.xmp`;
+    const sidecarPathWithExtExists = await this.storageRepository.checkFileExists(sidecarPathWithExt, constants.R_OK);
+    if (sidecarPathWithExtExists) {
+      await this.assetRepository.save({ id: asset.id, sidecarPath: sidecarPathWithExt });
+      return true;
+    }
+
+    const assetPath = path.parse(asset.originalPath);
+    const assetPathWithoutExt = path.join(assetPath.dir, assetPath.name);
+    const sidecarPathWithoutExt = `${assetPathWithoutExt}.xmp`;
+    const sidecarPathWithoutExtExists = await this.storageRepository.checkFileExists(
+      sidecarPathWithoutExt,
+      constants.R_OK,
+    );
+    if (sidecarPathWithoutExtExists) {
+      await this.assetRepository.save({ id: asset.id, sidecarPath: sidecarPathWithoutExt });
       return true;
     }
```

Comment:
You can use `Promise.all` to run both checks at the same time instead of sequentially. You can also assign the chosen sidecar path to a `sidecarPath` variable and use that when saving to avoid duplication.

Reply:
Updated